src/plugins/matrixops/reshape_operation.cpp

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) 2018 Bita Hasheminezhad
+// Copyright (c) 2018 Shahrzad Shirzad
+// Copyright (c) 2018 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/plugins/matrixops/reshape_operation.hpp>
+#include <phylanx/util/matrix_iterators.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+#include <hpx/util/iterator_facade.hpp>
+#include <hpx/util/optional.hpp>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    match_pattern_type const reshape_operation::match_data =
+    {
+        hpx::util::make_tuple("reshape",
+        std::vector<std::string>{"reshape(_1,_2)"},
+        &create_reshape_operation, &create_primitive<reshape_operation>,
+         R"(
+            a, newshape
+            Args:
+
+                a (array_like) : input array
+                newshape (integer or tuple of integers): The new shape should be
+                     compatible with the original shape (number of elements in
+                     both arrays are the same). If an integer, then the result
+                     will be a 1-D array of that length. The last parameter of
+                     the newshape can be -1. In this case, the value is inferred
+                     from the length of the array and remaining dimensions.
+
+            Returns:
+
+            Returns a new shape to an array without changing its data."
+            )") };
```

Comment:
In order for the `...__float` names etc. to be recognized by the compiler, the `match_data` needs to be passed a `true` as the second argument during construction.

Reply:
Thank you for your comment, I will correct this in the `eye` primitive. NumPy [reshape](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.reshape.html) does not have a `dtype`, so here, I only tried to extract the given type and produce the same thing in the new shape.

Reply:
I'm referring to https://github.com/STEllAR-GROUP/phylanx/pull/691/files#diff-be4e16b025144db3a166da2121df2dcaR58 where the current code tries to extract the `dtype` for the operation from the primitive's name.